tests/spark_autologging/test_pyspark_ml_autologging.py

```diff
@@ -0,0 +1,222 @@
+import importlib
+import inspect
+import pytest
+from collections import namedtuple
+from unittest import mock
+
+import mlflow
+from mlflow.utils import (
+    truncate_dict,
+)
+from mlflow.utils.validation import (
+    MAX_PARAM_VAL_LENGTH,
+    MAX_ENTITY_KEY_LENGTH,
+)
+
+from tests.autologging.fixtures import test_mode_off
+from tests.spark_autologging.utils import spark_session  # pylint: disable=unused-import
+from pyspark.ml.linalg import Vectors
+from pyspark.ml.regression import LinearRegression
+from mlflow.pyspark.ml import _should_log_model, _get_instance_param_map, _log_model_allowlist
+
+MODEL_DIR = "model"
+MLFLOW_PARENT_RUN_ID = "mlflow.parentRunId"
+
+
+@pytest.fixture(scope="module")
+def dataset_binary(spark_session):
+    yield spark_session.createDataFrame(
```

Comment:
I'd prefer using `return` if we don't run any teardown code:

```suggestion
    return spark_session.createDataFrame(
```